### PR TITLE
samples: nrf_desktop: Integrate DFU service

### PR DESCRIPTION
**Pull request by pdunaj**
Jira:DESK-303

Signed-off-by: Pawel Dunaj <pawel.dunaj@nordicsemi.no>

### CODE REVIEW

**Comment by pdunaj** [comment]
Tested on PCA20041. 
```
pdunaj@ubuntu:~/work/mcumgr$ sudo ./bin/mcumgr --conntype ble --connstring peer_name="Mouse nRF52" image upload /home/pdunaj/work/fw-nrfconnect-nrf/samples/nrf_desktop/build-pca20041/zephyr/zephyr_signed.bin -t 60
 261.15 KiB / 261.15 KiB [===============================================================================================================================================] 100.00% 1.84 KiB/s 2m21s
Done
```

This feature requires additional RAM memory - we need not only to give SMP service buffers but also increase MTU and workqueue stack.

I will try to optimize the RAM usage in the following commit.

File: samples/nrf_desktop/src/modules/ble_adv.c [hunk 1]
```diff
@@ -46,10 +46,13 @@ static const struct bt_le_adv_param adv_param_normal = {
 static const struct bt_data ad[] = {
 	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
 	BT_DATA_BYTES(BT_DATA_UUID16_ALL,
-#if CONFIG_BT_GATT_HIDS
+#if CONFIG_DESKTOP_HIDS_ENABLE
 			  0x12, 0x18,	/* HID Service */
 #endif
-			  0x0f, 0x18),	/* Battery Service */
+#if CONFIG_DESKTOP_BAT_ENABLE
```
Comment (Qbicz) [inline samples/nrf_desktop/src/modules/ble_adv.c]: `CONFIG_DESKTOP_BAS_ENABLE`

File: samples/nrf_desktop/src/services/smp.c [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include "img_mgmt/img_mgmt.h"
+#endif
+
+#define MODULE smp
+#include "module_state_event.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_BAS_LOG_LEVEL);
```
Comment (Qbicz) [inline samples/nrf_desktop/src/services/smp.c]: Please create `CONFIG_DESKTOP_SMP_LOG_LEVEL`

File: samples/nrf_desktop/src/modules/ble_adv.c (hunk 1 above)
Comment (pdunaj) [reply samples/nrf_desktop/src/modules/ble_adv.c]: Fixed

File: samples/nrf_desktop/src/services/smp.c (hunk 2 above)
Comment (pdunaj) [reply samples/nrf_desktop/src/services/smp.c]: Fixed

**Comment by pdunaj** [comment]
Hi @pizi-nordic , can you merge?

**Pull request merged**